functions/slack/app/setup.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2018, Google, LLC.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_slack_setup]
+
+package slack
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+
+	"google.golang.org/api/kgsearch/v1"
+	"google.golang.org/api/option"
+)
+
+type configuration struct {
+	ProjectID string `json:"PROJECT_ID"`
+	Token     string `json:"SLACK_TOKEN"`
+	Key       string `json:"KG_API_KEY"`
+}
+
+type attachment struct {
```

Comment:
Types should generally be next to where they are used. In this case, `attachment` and `Message` should be above `formatSlackMessage`. Otherwise, you have to look back and forth between files to know what this is.